src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -261,6 +277,11 @@ export class TerminalInstance implements ITerminalInstance {
 			if (TabFocus.getTabFocusMode() && event.keyCode === 9) {
 				return false;
 			}
+
+			if (platform.isWindows && event.keyCode === 13 /* ENTER */) {
+				this._checkWindowShell.fire();
```

Comment:
I think we actually want to use this instead of bringing in the overhead of an event/emitter https://github.com/Microsoft/vscode/blob/e49ae72a37b22a7e7f2ed3a6bf052076e9a283d9/extensions/git/src/contentProvider.ts#L42